File: src/main/java/fr/lewon/bot/runner/bot/operation/OperationResult.java
package fr.lewon.bot.runner.bot.operation;

public class OperationResult {

    private boolean success;
    private String message;
    private Object content;

    public OperationResult(boolean success, String message, Object content) {
        this.success = success;
        this.message = message;
        this.content = content;
    }

    public boolean isSuccess() {
        return this.success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getMessage() {
        return this.message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Object getContent() {
        return this.content;
    }

    public void setContent(Object content) {
        this.content = content;
    }

    @Override
    public String toString() {
        return "OperationResult{" +
                "success=" + success +
                ", message='" + message + '\'' +
                ", content=" + content +
                '}';
    }
}

File: src/main/kotlin/fr/lewon/bot/runner/bot/operation/OperationResult.kt
package fr.lewon.bot.runner.bot.operation

class OperationResult(var isSuccess: Boolean = false, var message: String = "", var resultType: ResultType = ResultType.STRING, var content: Any? = "") {

    override fun toString(): String {
        return "OperationResult(isSuccess=$isSuccess, message=$message, content=$content)"
    }

    companion object {

        fun empty(isSuccess: Boolean, message: String): OperationResult {
            return OperationResult(isSuccess, message)
        }

        fun ofString(isSuccess: Boolean, message: String, content: String): OperationResult {
            return OperationResult(isSuccess, message, ResultType.STRING, content)
        }

        fun ofList(isSuccess: Boolean, message: String, content: List<OperationResult>): OperationResult {
            return OperationResult(isSuccess, message, ResultType.LIST, content)
        }

        fun ofObject(isSuccess: Boolean, message: String, content: Any?): OperationResult {
            return OperationResult(isSuccess, message, ResultType.OBJECT, content)
        }

        fun ofGraph(isSuccess: Boolean, message: String, content: Any?): OperationResult {
            return OperationResult(isSuccess, message, ResultType.GRAPH, content)
        }

        fun ofImageUrl(isSuccess: Boolean, message: String, content: String): OperationResult {
            return OperationResult(isSuccess, message, ResultType.IMAGE_URL, content)
        }

    }

}
